rust/src/settings_manager.rs

```diff
@@ -1,72 +1,133 @@
-use std::sync::Mutex;
+use std::fmt;
 
-use yaml_rust::{Yaml, YamlLoader};
+use serde::{
+    de::{self, Unexpected},
+    Deserialize,
+};
 
-const SOURCE: &str = include_str!("../../settings.yaml");
+lazy_static::lazy_static! {
+    static ref SETTINGS: Vec<Setting> = {
+        serde_yaml::from_str(include_str!("../../settings.yaml"))
+            .expect("Could not parse settings.yaml")
+    };
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct Setting {
+    name: String,
+
+    group: String,
+
+    #[serde(rename = "type")]
+    kind: Option<SettingKind>,
```

Comment:
Some settings in group cn0_est have no type, I feel like at the very least we should add those in

Reply:
Yes, we should

Reply:
These are the fields without types, any idea who we should ping to get the values?

```
- group: cn0_est
  name: alpha
  expert: true
  Description: cn0 estimation parameter.
  readonly: false

- group: cn0_est
  name: nbw
  expert: true
  Description: cn0 estimation parameter.
  readonly: false

- group: cn0_est
  name: scale
  expert: true
  Description: cn0 estimation parameter.
  readonly: false

- group: cn0_est
  name: cn0 shift
  expert: true
  Description: cn0 estimation parameter.
  readonly: false

- group: cn0_est
  name: cn0_shift
  expert: true
  Description: cn0 estimation parameter.
  readonly: false

- group: cn0_est
  name: cutoff
  expert: true
  Description: cn0 cutoff.
  readonly: false
```

Reply:
> 
> 
> These are the fields without types, any idea who we should ping to get the values?
> 
> ```
> - group: cn0_est
>   name: alpha
>   expert: true
>   Description: cn0 estimation parameter.
>   readonly: false
> 
> - group: cn0_est
>   name: nbw
>   expert: true
>   Description: cn0 estimation parameter.
>   readonly: false
> 
> - group: cn0_est
>   name: scale
>   expert: true
>   Description: cn0 estimation parameter.
>   readonly: false
> 
> - group: cn0_est
>   name: cn0 shift
>   expert: true
>   Description: cn0 estimation parameter.
>   readonly: false
> 
> - group: cn0_est
>   name: cn0_shift
>   expert: true
>   Description: cn0 estimation parameter.
>   readonly: false
> 
> - group: cn0_est
>   name: cutoff
>   expert: true
>   Description: cn0 cutoff.
>   readonly: false
> ```

These don't seem to exist in any code anywhere anymore: https://github.com/search?q=org%3Aswift-nav+cn0_est&type=code -- so we can either remove or ignore these